src/libstd/sys/unix/fs.rs

```diff
@@ -269,6 +269,9 @@ impl File {
             libc::open(path.as_ptr(), flags, opts.mode)
         }));
         let fd = FileDesc::new(fd);
+        // Even though we open with the O_CLOEXEC flag, still set CLOEXEC here,
```

Comment:
hm, since @alexcrichton pointed out that there are tests for this, would it be appropriate to add `[cfg(not(any(target_os = "freebsd", target_os = "dragonfly", ...)))]`? perhaps pedantic, but it would assert that constants added here are correct